### ISSUE TITLE
Express Middleware

### ISSUE THREAD

**Issue by rbriois** [lukeautry/tsoa#103]
Hi folks, 

I encounter some problems with the routes generated by tsoa for express. 
In fact, I declared a middleware that i want to be used after the request was processed by routes. 

In the generated routes, I end up in the function bellow. In the case of you have data returned, the next function is never called. 
```typescript
function promiseHandler(promise: any, statusCode: any, response: any, next: any) {
        return promise
            .then((data: any) => {
                if (data) {
                    response.json(data);
                    response.status(statusCode || 200);
                } else {
                    response.status(statusCode || 204);
                    response.end();
                }
            })
            .catch((error: any) => next(error))
    }
```

Is there a particular reason for this behavior or is this just a miss ? 

In case, there is no reason, would you accept a PR to correct this behavior ? 

Cheers

**Comment by lukeautry** [comment]
That is definitely a miss. PR is welcome, thanks.

**Comment by rbriois** [comment]
Will do it within a few days (with unit tests) 
Thanks for your quick answer

**Comment by isman-usoh** [comment]
Fix 804ed90db71b8565255c42fcea09b5ad11fcc10b
